neuraxle/steps/util.py

```diff
@@ -517,7 +511,45 @@ def get_cache_path_for(self, data_input) -> str:
         raise NotImplementedError()
 
 
-class PickleValueCachingWrapper(BaseValueCachingWrapper):
+class OutputTransformerMixin:
+    """
+    Base output transformer step that can modify data inputs, and expected_outputs at the same time.
+    """
+
+    def handle_transform(self, data_container: DataContainer) -> DataContainer:
+        """
+        Handle transform by updating the data inputs, and expected outputs inside the data container.
+
+        :param data_container:
+        :return:
+        """
+        di_eo = (data_container.data_inputs, data_container.expected_outputs)
+        new_data_inputs, new_expected_outputs = self.transform(di_eo)
```

Comment:
For this `OutputTransformerMixin` class, we might want another version that is a `ZipOutputTransformerMixin` which calls zip like `zip(di_eo)` to transpose the tuple. This is not needed yet so my present comment is just an idea.

Reply:
outdated